### ISSUE TITLE
False-positive `Lint/LambdaWithoutLiteralBlock` when using `itself`

### ISSUE THREAD

**Issue by WaKeMaTTa** [rubocop/rubocop#10489]
## Expected behavior

Rubocop shouldn't complain.

## Actual behavior

Rubocop complains with `Lint/LambdaWithoutLiteralBlock`.

## Steps to reproduce the problem

**`my_file.rb`**
```ruby
service = lambda(&:itself)
service.call("Hello")
```

**`$ bundle exec rubocop my_file.rb`**
```bash
my_file.rb:10:1: W: [Correctable] Lint/LambdaWithoutLiteralBlock: lambda without a literal block is deprecated; use the proc without lambda instead.
```

## RuboCop version

```
$ [bundle exec] rubocop -V
1.26.1 (using Parser 3.1.1.0, rubocop-ast 1.16.0, running on ruby 3.0.1 x86_64-linux)
  - rubocop-performance 1.13.3
  - rubocop-rails 2.14.2
  - rubocop-rspec 2.9.0
```
